### PR TITLE
Adds FreeSASA adapter

### PR DESCRIPTION
**Pull request by bp-kelley**
Adds an adapter to the FreeSASA (Solvent Accessible Surface Area) library:

 http://freesasa.github.io

### CODE REVIEW

**Comment by greglandrum** [comment]
Looks mostly good.
We can do the python wrapper once this is merged.

The appveyor failure is probably worth looking at though. If there are going to be timeout problems with this code we should probably disable the freesasa build on appveyor

**Comment by thegodone** [comment]
Very good do you want i add it in my presantion related to 3D descriptors?

Envoyé de mon iPhone

**Comment by greglandrum** [comment]
Hard to say at the moment. I think it's likely this will be merged in time for the next release, but that's not sure yet.

**Comment by thegodone** [comment]
ok I can add it at the last time if needed

2017-09-13 8:53 GMT+02:00 Greg Landrum <notifications@github.com>:

**Comment by bp-kelley** [comment]
@greglandrum I couldn't figure out the queryAtom PythonAPI, could you take a look?

**Comment by greglandrum** [comment]
Yep, will look tomorrow morning

**Comment by bp-kelley** [comment]
Remind me to turn appveyor off before the final merge.

**Comment by bp-kelley** [comment]
@greglandrum Sorted it out.  Passing in null pointers as python args needs to be done as follows:

```
python::args("query") = python::object()
```

File: External/FreeSASA/RDFreeSASA.h [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    FreeSASA identified Classes end up in atom.getProp(common_properties::Atom::FreeSASAClass)
+       returns false if no atoms could be classified
+*/       
+bool classifyAtoms(RDKit::ROMol &mol, std::vector<double> &radii,
+                   const FreeSASA::SASAOpts &opts = SASAOpts());
+
+//! calculate all atom contributions
+/*!
+  SASA data is stored in atom.getProp(common_properites::Atom::SASA);
+ARGUMENTS:
```
Comment (greglandrum) [inline External/FreeSASA/RDFreeSASA.h]: This commenting style is perfectly readable, but I don't think doxygen will handle it correctly.
If that's desirable (I don't think it's critical), here's the style used elsewhere: https://github.com/rdkit/rdkit/blob/master/Code/GraphMol/ROMol.h#L180

File: External/FreeSASA/Wrap/rdFreeSASA.cpp [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        .value("OONS", FreeSASA::SASAOpts::OONS)
+        .export_values();
+
+    python::enum_<FreeSASA::SASAOpts::Classes>("SASAClass")
+        .value("Unclassified", FreeSASA::SASAOpts::Unclassified)
+        .value("APolar", FreeSASA::SASAOpts::APolar)
+        .value("Polar", FreeSASA::SASAOpts::Polar)
+        .export_values();
+
+    python::class_<FreeSASA::SASAOpts>("SASAOpts", docString.c_str(),
```
Comment (greglandrum) [inline External/FreeSASA/Wrap/rdFreeSASA.cpp]: It looks like this class doesn't expose any methods or data members, which feels weird.
How about exposing the algorithm and classifier using `def_readwrite()`?

File: External/FreeSASA/RDFreeSASA.h (hunk 1 above)
Comment (bp-kelley) [reply External/FreeSASA/RDFreeSASA.h]: Whups, I copied the style from boost:python! I'll fix it.

**Comment by greglandrum** [comment]
The appveyor failures are not the usual timeouts. They seem to be compile errors:
https://ci.appveyor.com/project/greglandrum/rdkit/build/2016.03.1.dev1960#L825

**Comment by bp-kelley** [comment]
I can patch these for now and push the changes upstream.

**Comment by bp-kelley** [comment]
@greglandrum Patched out windows, will update when I get the patch working.

**Comment by greglandrum** [comment]
I've tested on both Linux and the Mac and think this is ready to be merged; we will likely do more tuning as we gain experience with it, but that's no reason to delay the merge

**Pull request merged**